src/test/diagnostics.test.ts

```diff
@@ -0,0 +1,190 @@
+import * as assert from 'assert';
+import * as vscode from 'vscode';
+
+// Shared setup for all diagnostic tests
+suiteSetup(async () => {
+    const extension = vscode.extensions.getExtension('andy9a9.vscode-devicetree');
+    if (extension && !extension.isActive) {
+        await extension.activate();
+    }
+});
+
+/**
+ * Helper function to get diagnostics for a document
+ */
+async function getDiagnostics(content: string): Promise<vscode.Diagnostic[]> {
+    const doc = await vscode.workspace.openTextDocument({
+        language: 'dts',
+        content: content
+    });
+
+    // Wait for diagnostics to be computed
+    await new Promise(resolve => setTimeout(resolve, 600));
+
+    const diagnostics = vscode.languages.getDiagnostics(doc.uri);
+    return diagnostics;
+}
+
+suite('DTS Diagnostics - Line Length', () => {
+    test('Should not warn for lines within limit', async () => {
+        const input = '/ {\n\tmodel = "Test";\n};';
+        const diagnostics = await getDiagnostics(input);
+        assert.strictEqual(diagnostics.length, 0);
+    });
+
+    test('Should warn for lines exceeding 80 characters', async () => {
+        const input = '/ {\n\tmodel = "This is a very long string that definitely exceeds the maximum line length of 80 characters";\n};';
+        const diagnostics = await getDiagnostics(input);
+        assert.ok(diagnostics.length > 0);
```

Comment:
Any reason why you are not using `assert.strictNotEqual(diagnostics.length, 0)`?
This way, you would use the same pattern whether you expect errors or not.